### PR TITLE
Webpack 4, bundle optimizations, blue-green deployment

### PR DESCRIPTION
**Pull request by inyono**
### Added

- Dynamically import the new editor and its renderer. This leads to a much smaller bundle size for the initial load.
- Automated blue-green deployment of athene2-assets and server side renderer for new editor
- Automated deployment of Cloudflare workers
- Log version to browser console
- Add a content hash of length 8 to file names of chunks and assets (e.g. fonts and images)

### Changed

- Upgrade to Webpack 4

### Fixed

- Load all plugins in render server for new editor
- When server side rendering fails, still pass state down to client for improved debugging

### Breaking Changes

- Removed `CommonsChunkPlugin`, i.e. there are no more `commons.js` and `common.css`
- Renamed `editor` to `legacy-editor`, i.e. one has to include `legacy-editor.js` resp. `legacy-editor.css` instead of `editor.js` resp. `editor.css`
- Asset bundle moved to `https://packages.serlo.org/athene2-assets@a/` resp. `https://packages.serlo.org/athene2-assets@b/` (blue-green deployment)
- Google Cloud Function moved to `https://europe-west1-serlo-assets.cloudfunctions.net/editor-renderer-a` resp. `https://europe-west1-serlo-assets.cloudfunctions.net/editor-renderer-b` (blue-green deployment)

### CODE REVIEW

File: src/editor/editor/index.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const initEntityEditor = (...args: unknown[]) => {
+  // prettier-ignore
+  return import(
```
Comment (inyono) [inline src/editor/editor/index.ts]: @Knorrke dynamically import the entity editor

File: src/editor/server-render.js [hunk 2]
```diff
@@ -33,16 +38,19 @@ export function render(input, callback) {
...
-    const oryState = data['cells'] ? data : converter(data)
+    const state = data['cells'] ? data : converter(data)
 
-    const output = renderToString(
-      <HtmlRenderer state={oryState} plugins={createRenderPlugins()} />
-    )
+    try {
+      const children = renderToString(
+        <HtmlRenderer
+          state={oryState}
```
Comment (Knorrke) [inline src/editor/server-render.js]: this should be state now

File: src/main/index.js [hunk 3]
```diff
@@ -186,14 +185,69 @@ const init = $context => {
...
+          )
+          return $editable.closest('article').length
+            ? $editable.closest('article')
+            : $all.find('#content-layout article')
+        }
+
+        const $target = getEditedArticle($('body'))
+        const $dataArticle = getEditedArticle($(data))
+        $target.html($dataArticle.html())
+      })
```
Comment (Knorrke) [inline src/main/index.js]: This ajax call is here twice. Is this intentional or a mistake?

File: src/editor/server-render.js (hunk 2 above)
Comment (inyono) [reply src/editor/server-render.js]: Thanks

File: src/main/index.js (hunk 3 above)
Comment (inyono) [reply src/main/index.js]: Nope, thanks. And I also saw the `$('#loading').hide()` which I can just move down and fix the mentioned issue in the comment :)

**Comment by inyono** [comment]
@Knorrke: can you please review the automated deployment script (mostly done, except for additional logs (e.g. webpack) and success checks (e.g. GET bundle)) https://github.com/serlo-org/athene2-assets/blob/ca8e94663b41c7ee1c24da90621e5b9fc00c3433/scripts/deploy.ts

**Pull request merged**